x-pack/plugins/ml/public/application/components/annotations/annotations_table/annotations_table.js

```diff
@@ -255,18 +257,18 @@ export class AnnotationsTable extends Component {
 
     // if the annotation is at the series level
     // then pass the partitioning field(s) and detector index to the Single Metric Viewer
-    if (_.has(annotation, 'detector_index')) {
+    if (annotation.detector_index !== undefined) {
```

Comment:
Not sure if it's important here, but [has](https://lodash.com/docs/4.17.15#has) is not the same as check for undefined. Would be safer to replace with `hasOwnProperty`

Reply:
I agree that `has` is not the same as checking for `undefined`, however the use of `has` in our code was not as a replacement for `hasOwnProperty`, it was only ever used to check whether a property had been set, and was not considering whether the property existed on the prototype chain.
In all other similar checks in our code base we use an `undefined` check, so for consistency i'd like to keep these changes.
but this is a good thing to point out!